src/main/drivers/io.c

```diff
@@ -238,19 +238,25 @@ void IOToggle(IO_t io)
 // claim IO pin, set owner and resources
 void IOInit(IO_t io, resourceOwner_e owner, uint8_t index)
 {
+    if (!io)
```

Comment:
It's called from `serialUART` in `serial_uart_stm32f4xx.c`. The other option would be to have the check there (and in the F1 / F3 / F7 version) or to not support serial ports with only Tx or Rx pin assigned. IMO, it makes sense to support this, as it makes sense to have an Rx only UART for receivers, as it uses fewer pins on the STM32 and less PCB space.